pkg/3scale/amp/operator/upgrade.go

```diff
@@ -727,6 +736,72 @@ func (u *UpgradeApiManager) ensurePodTemplateLabels(desired *appsv1.DeploymentCo
 	return updated, nil
 }
 
+func (u *UpgradeApiManager) upgradePVCSizes() (reconcile.Result, error) {
+	if !u.apiManager.IsExternalDatabaseEnabled() {
+		if u.apiManager.IsSystemPostgreSQLEnabled() {
+			err := u.updatePVCSize(component.SystemPostgreSQLPVCName, resource.MustParse("10Gi"))
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+
+		if u.apiManager.IsSystemMysqlEnabled() {
```

Comment:
I think this would not be totally correct.

IsSystemMysqlEnabled is:

```
func (apimanager *APIManager) IsSystemMysqlEnabled() bool {
	return !apimanager.IsExternalDatabaseEnabled() &&
		apimanager.Spec.System.DatabaseSpec != nil &&
		apimanager.Spec.System.DatabaseSpec.MySQL != nil
}
```

This checks whether System MySQL is explicitely enabled. There's another case where MySQL is enabled, which is the default case (where system mysql usage is implicit). In the default case no spec.system.databasespec.mysql is set

Reply:
Right.

I fixed updating the `IsSystemMysqlEnabled` method to 
```
func (apimanager *APIManager) IsSystemMysqlEnabled() bool {                
    // Mysql is the default one                                            
    // If it happens to be more DB's, they should also be included         
    return !apimanager.IsExternalDatabaseEnabled() &&                      
        !apimanager.IsSystemPostgreSQLEnabled()                            
}                                                                          
```